tsl/src/compression/compression.c

```diff
@@ -163,30 +166,47 @@ truncate_relation(Oid table_oid)
 	List *fks = heap_truncate_find_FKs(list_make1_oid(table_oid));
 	/* Take an access exclusive lock now. Note that this may very well
 	 *  be a lock upgrade. */
-	Relation rel = relation_open(table_oid, AccessExclusiveLock);
+	Relation rel = table_open(table_oid, AccessExclusiveLock);
+#if PG12_LT
 	MultiXactId minmulti;
+#endif
 	Oid toast_relid;
 
 	/* Chunks should never have fks into them, but double check */
 	if (fks != NIL)
 		elog(ERROR, "found a FK into a chunk while truncating");
 
 	CheckTableForSerializableConflictIn(rel);
-
+#if PG12_LT
 	minmulti = GetOldestMultiXactId();
+#endif
 
-	RelationSetNewRelfilenode(rel, rel->rd_rel->relpersistence, RecentXmin, minmulti);
+	RelationSetNewRelfilenode(rel,
+							  rel->rd_rel->relpersistence
+#if PG12_LT // FIXME @Matve review d25f519, check if this is right
```

Comment:
@cevian @JLockerman Do you know anything about this comment? cc @erimatnor

Reply:
I'll let @cevian chime in, but I think this is what it is right now.

Reply:
that was asking to check the compat against postgres commit https://github.com/postgres/postgres/commit/d25f519107bff602e1ebc81853fe592d020c118d to make sure it was correct

Reply:
@JLockerman Since the new `RelationSetNewRelfilenode` doesn't give the option to specify neither `freezeXid`, nor  `minmulti` I guess the question is only if the old code is correct? Or, am I missing something?

FWIW, the new `RelationSetNewRelfilenode`  reads those values internally depending on relation kind, using the function `table_relation_set_new_filenode`.

Reply:
@erimatnor mostly I wanted to be sure we weren't accidentally introducing a visible change across versions, since in the old one we ensured that the minimum transaction ID wasn't too low, and now we're relying on postgres. Since the function is guaranteed to be transactional, I don't think there should be an issue, though it is possible we were being more conservative than postgres would be.